addons/event/models/event_registration.py

```diff
@@ -148,12 +149,53 @@ def _synchronize_partner_values(self, partner, fnames=None):
 
     @api.model_create_multi
     def create(self, vals_list):
+        for vals in vals_list:
+            vals['utm_campaign_id'] = vals.get('utm_campaign_id') or self.get_utm_params(['utm_campaign_id']).get('utm_campaign_id')
+            vals['utm_source_id'] = vals.get('utm_source_id') or self.get_utm_params(['utm_source_id']).get('utm_source_id')
+            vals['utm_medium_id'] = vals.get('utm_medium_id') or self.get_utm_params(['utm_medium_id']).get('utm_medium_id')
+
         registrations = super(EventRegistration, self).create(vals_list)
         if registrations._check_auto_confirmation():
             registrations.sudo().action_confirm()
 
         return registrations
 
+    def get_utm_params(self, fields):
+        values = {}
+
+        # We ignore UTM for salesmen, except some requests that could be done as superuser_id to bypass access rights.
+        if not self.env.is_superuser() and self.env.user.has_group('sales_team.group_sale_salesman'):
+            return values
+
+        for _, field_name, cookie_name in self.utm_tracking_fields():
+            if field_name in fields:
+                field = self._fields[field_name]
+                value = False
+                if request:
+                    # ir_http dispatch saves the url params in a cookie
+                    value = request.httprequest.cookies.get(cookie_name)
+                # if we receive a string for a many2one, we search/create the id
+                if field.type == 'many2one' and isinstance(value, str) and value:
+                    Model = self.env[field.comodel_name]
+                    records = Model.search([('name', '=', value)], limit=1)
+                    if not records:
+                        if 'is_website' in records._fields:
+                            records = Model.create({'name': value, 'is_website': True})
+                        else:
+                            records = Model.create({'name': value})
+                    value = records.id
+                if value:
+                    values[field_name] = value
+        return values
```

Comment:
Can we put this in default_get? Would avoid the 'vals or ...' in the create and it's more consistent with the mixin